crescent/commands/errors.py

```diff
@@ -0,0 +1,52 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Callable, Generic, Protocol, Type, TypeVar
+
+from crescent.internal.meta_struct import MetaStruct
+
+if TYPE_CHECKING:
+    from crescent.context import Context
+    from crescent.internal.app_command import AppCommandMeta, CommandCallback
+    from crescent.typedefs import CommandOptionsT
+
+
+ERROR = TypeVar("ERROR", bound=Exception, contravariant=True)
+
+
+class ErrorHandlerProto(Protocol, Generic[ERROR]):
+    async def __call__(
+        self,
+        /,
+        *,
+        exc: ERROR,
+        ctx: Context,
+        command: MetaStruct[CommandCallback, AppCommandMeta],
+        options: CommandOptionsT,
+    ) -> None:
+        ...
+
+
+def catch(
+    exception: Type[ERROR],
```

Comment:
I changed my mind about this. It definitely needs to be `*exceptions`.

Reply:
Found a middle ground using overloads. If you pass one exception, you get the type checking stuff, but it's disabled if you pass multiple.